<script>
    function outside(x) {
        function inside(y) {
            return x + y;
        }
        return inside;
    }

    // 내부 함수는 클로저를 형성하기 때문에 외부함수를 호출하고, 외부 및 내부함수 모두에 인수 지정할 수 있음
    // inside가 반환될 때 외부 함수의 인수 x가 보존됨 = closure 현상
    fn_inside = outside(3); result = fn_inside(5);
    console.log("fn_inside = outside(3); result = fn_inside(5); : " + result);
    result = outside(3)(5);
    console.log("outside(3)(5) : " + result)
</script>